PMDMMO-common/src/com/darkxell/common/ai/visibility/Visibility.java

```diff
@@ -0,0 +1,112 @@
+package com.darkxell.common.ai.visibility;
+
+import java.util.HashSet;
+
+import com.darkxell.common.ai.AI;
+import com.darkxell.common.dungeon.data.FloorData;
+import com.darkxell.common.dungeon.floor.Tile;
+import com.darkxell.common.dungeon.floor.room.Room;
+import com.darkxell.common.item.ItemEffects;
+import com.darkxell.common.pokemon.DungeonPokemon;
+import com.darkxell.common.util.Direction;
+
+public class Visibility {
+
+    /** Reference to parent AI. */
+    public final AI ai;
+    /** The Tiles that this Pokemon can currently see. */
+    private HashSet<Tile> currentlyVisible = new HashSet<>();
+    /** The Tiles that were seen with an Item on it. */
+    private HashSet<Tile> itemTiles = new HashSet<>(); // TODO has to update when Items are moved
+    /** The Tiles that were already seen. */
+    private HashSet<Tile> seenTiles = new HashSet<>();
+
+    public Visibility(AI ai) {
+        this.ai = ai;
+    }
+
+    /** @return The Tiles this Pokemon can currently see from the Tile it's at. */
+    public HashSet<Tile> currentlyVisibleTiles() {
+        return new HashSet<>(this.currentlyVisible);
+    }
+
+    /** @return <code>true</code> if the input Tile has been seen in this Floor. */
+    public boolean hasSeenTile(Tile tile) {
+        return this.seenTiles.contains(tile);
+    }
+
+    /**
+     * @return <code>true</code> if the Item on the input Tile is visible by this Pokemon (doesn't check if there
```

Comment:
I feel like `isX` functions should have their docs pose a question, eg:

```
Would an item on the said tile be visible to this AI? (ignores actual status)
```

Not a huge deal, just something to think about. Can ignore if you have any reason.

Also, I feel that we should prefer `{@code }` over `<code />` in javadocs.

Also, maybe we should move away from using the term Pokemon when possible for copyright reasons, and use eg Species or AI etc instead.